src/main/java/uk/gov/hmcts/darts/event/service/handler/base/EventHandlerBase.java

```diff
@@ -69,6 +69,19 @@ protected CreatedHearingAndEvent createHearingAndSaveEvent(DartsEvent dartsEvent
             caseNumber,
             dartsEvent.getDateTime().toLocalDate()
         );
+        UserAccountEntity currentUser = authorisationApi.getCurrentUser();
+
+        if (hearingEntity.getCourtCase().getCreatedBy() == null) {
```

Comment:
probably better to update retrieveOrCreateHearing to accept the userAccountEntity

Reply:
I am going to leave that as an improvement , that's going to change a lot of different places and will require a lot more work imo.

Reply:
Hi @HMCTSDeveloper please could you create a new ticket for this improvement? We will want to do it, but happy for it to be separate to this PR.

Reply:
> Hi @HMCTSDeveloper please could you create a new ticket for this improvement? We will want to do it, but happy for it to be separate to this PR.

https://tools.hmcts.net/jira/browse/DMP-2322